### DISCUSSION TITLE
Anyone looking for a web dashboard?

### DISCUSSION THREAD

**Discussion by brutella** [brutella/hkknx-public#242] [Ideas]
I'm experimenting with a dashboard to visualise the configured accessories in a browser.

I'm currently at a point where the accessories appear similar to HomeKit. The accessories are grouped by rooms. Lights, outlets, switches and blinds appear in a square tile. Tapping/clicking on the tile toggles its state. Sensors like temperature, humidity, motion, occupancy sensors are non-interactive and appear above the tiles and show their values.

Now I have to decide if it makes sense to keep working on this feature.

I think a dashboard might be useful in the following use cases.

**Creating a dashboard for specific people**
In HomeKit you can share a home with other people. But there is no way to restrict access to only specific accessories. This means that all people have access to all accessories. This might not be useful because you may not want to give access to your ventilation system to your kids. With a dashboard for specific people, you can restrict access to certain accessories.

**You don't want to depend on iOS** 
hkknx is currently only useful for people with iOS devices. If there is a way to "recreate" the HomeKit user-interface in a dashboard, you can use hkknx also with Android devices. This makes hkknx useable for non-Apple users.

**Show non-HomeKit accessories in a dashboard**
HomeKit only supports specific accessories and sensors. As noted in #95 there might be a need to show other sensors values as well. A web dashboard could show any accessory/sensor.

Does anyone actually have a need for a dashboard?

[a comment by an automated account is not shown]

**Comment by mirkolenz** [comment]
I have created a custom dashboard in Home Assistant, but I still think that an automatically generated (and visually pleasing) dashboard would be a huge benefit for hkknx. Besides supporting devices that cannot be added to HomeKit, it may also be a straightforward way to include statistics of the devices or the KNX bus itself (like proposed in #241).

**Comment by fidelio-1** [comment]
I see a huge potential in dashboards that could be well integrated with hkknx, so to say homekit (or without) and knx.
hkknx with a dashboard, could fulfill my needs where today Homekit alone cannot make it.
These are specifically:
- online and archives of measuring values like energy, power consumption, kWh "Ganglinien" over days, weeks, etc.
  in knx thease are 4byte values - I do not see that this is supported in hkknx today. (e.g. also operating hours).
- combined with a storage concept (e.g. influx)
- combined with a gateway approach (e.g. Modbus)
- also for temperatures, humidity etc.  
- max / min values by software
The initial reasons brutella gave are meaningful also to me (e.g. subset of functions for kids); however my "killer application" would be above..
I would also see a need for customized and fancy visualization of complete aggregates (e.g. Lüftung) where status switches can change e.g. the mode of the aggregate, just to give an example.

**Comment by Llanzelot** [comment]
Ich suche schon seit Ewigkeiten ein gutes Dashboard. Ob das nun eine App ist (Home+) oder ein Web-Basierte Geschichte (z.B. aus HKKNX heraus vom Raspi im lokalen Netz zur Verfügung gestellt) ist mir dabei egal. Web hat den Vorteil, dass man das Dashboard auch mal auf dem Smart TV anzeigen lassen kann.

Was mir bisher bei allen Lösungen gefehlt hat ist die Integration von **Kalender** und **Wetter**. Wenn ich morgens aus dem Haus gehe würde ich gerne wissen, ob ich die Biotonne mit rausschieben muss (kalendereintrag), ob ein Geburtstag oder Arzttermin ansteht (Kalendereintrag), ob ich einen Regenschirm/Jacke mitnehmen muss oder die Rolläden runterfahren sollte (Wetter).

Zwar gibt es Lösungen wie etwa Smart Mirror etc. die zwar Kalender und Wetter anzeigen können - allerdings fehlt da wiederum der Homekit Zugriff.

Vor einigen Jahren sah Homewall von Aaron Pierce wie eine vielversprechende Lösung aus, aber da kam leider nie etwas raus. =/

**Comment by brutella** [comment]
@Llanzelot Für Kalendar und Wetter gibt es ja schon gute Lösungen am iPad. Wenn du bspw. im Splitscreen die Home-App und Kalendar-App verwendest, hast ja alles gut im Blick. Oder wenn du die Fantastical-App verwendest, siehst du auch noch deine Erinnerungen neben den Kalendareinträgen.

Mit dem iPad bist du da sehr flexibel und kannst ausserdem noch Musik abspielen, im Web surfen, die Uhrzeit ablesen, ...

*Edit: Ich sehe gerade, dass Fantastical auch eine [Wetteransicht](https://flexibits.com/fantastical#weather-section) bereitstellt. Nice ;)*

---

Bei dem Dashboard würde ich mich hauptsächlich auf die Haussteuerung konzentrieren.

**Reply by Llanzelot** [reply]
Das ist wahr! das iPad kann Wetter und Kalender im Prinzip schon ganz gut.
Allerdings hängt man es als Dashboard ja gerne im "geführten Zugriff" auf - eine feste App, die immer angezeigt wird, wenn der Bildschirm angeht. Ohne Sperrbildschirm und irgendwelche Gesten.

Welche App das ist, ist hier auch wieder egal. Home+ oder Browser funktioniert beides.

Das hat in Verbindung mit KNX/HKKNX den netten Vorteil, dass das man das iPad mittels Bewegungsmelder und Notification aufwecken kann. Dann keuchtet es nicht permanent vor sich hin, sondern wirklich nur, wenn jemand den Flur/Wohnzimmer etc betritt.

**Comment by fidelio-1** [comment]
Ich bin da auch eher bei Llancelot.
Etwas Schönes Integriertes wäre cool und fehlt. Ein Live Dashboard, welches
verschiedene Widgets / Sichten ermöglicht zu kombinieren mit Daten aus der
hkknx Welt und Wetter/Kalender.


Llanzelot ***@***.***> schrieb am Mi. 11. Jan. 2023 um 15:26:

**Comment by Llanzelot** [comment]
Da ich das Internet wirklich rauf und runter gesucht habe, vermute ich auch, dass das die größere Marktlücke ist.
Für Smart Home/ Wetter/ Kalender als Paket gibt es keine gute Lösung. Probiert habe ich bisher folgendes.

- Smart Mirror (Raspi Server) Wetter geht, Apple Kalender nur über Umwege und auch dann noch sporadisch, Kein Homekit
- FHEM (Raspi Server) funktioniert nur als Bastellösung und sieht nicht unbedingt gut aus.
- PowerHouz (App) verspricht viel, funktioniert aber nicht
- HomeWatch (App) funktioniert, bietet aber nicht den Homekit-Zugriff, den ich gerne hätte (Szenen funktionieren, Lichter an/aus geht, Rolläden sind quasi nicht funktional). Ist aber aktuell meine Lösung

**Comment by mbrockeu** [comment]
Kann man dann in dem Dashboard nur die Geräte sehen welche in der Bridge sind oder alle in HomeKit?

**Reply by brutella** [reply]
Nur die eigenen Geräte werden angezeigt. Auf andere HomeKit-Geräte bekommt man keinen Zugriff – geht technisch gar nicht.

**Reply by mbrockeu** [reply]
ok habe ich mir schon fast gedacht. Dann wäre es natürlich schön, wenn man mehrere Bridges hätte die wenigstens zusammenfassen kann. Siehe auch meine angelegte Idee :-)
